src/Material.cpp

```diff
@@ -299,9 +299,9 @@ FP_PRECISION* Material::getDifHat() {
 
   if (_dif_hat == NULL) {
 
-    _dif_hat = new FP_PRECISION[4*_num_groups];
+    _dif_hat = new FP_PRECISION[NUM_SURFACES*_num_groups];
 
-    for (int e = 0; e < 4*_num_groups; e++)
+    for (int e = 0; e < NUM_SURFACES*_num_groups; e++)
```

Comment:
This is just me being picky and I think this would be more relevant for a different PR, but it seems like this loop would be more intuitive if broken up into two for one over surfaces and one over energies (and if `_dif_hat` had a 2D indexing macro accepting both surface and energy group indices).